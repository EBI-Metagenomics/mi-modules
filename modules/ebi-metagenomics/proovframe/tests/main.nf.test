nextflow_process {

    name "Test Process PROOVFRAME"
    script "../main.nf"
    process "PROOVFRAME"

    tag "modules"
    tag "modules_nfcore"
    tag "proovframe"

    test("mytest1 - faa_fasta") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/emales.faa", checkIfExists: true),
                    file("${moduleDir}/tests/fixtures/emales-reads-sim.fa", checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }  // Adding the version assertion
                // Additional assertions can be added here if necessary
            )
        }
    }
}
