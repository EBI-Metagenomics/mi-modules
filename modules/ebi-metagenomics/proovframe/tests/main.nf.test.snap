{
    "mytest1 - faa_fasta": {
        "content": [
            [
                "versions.yml:md5,e53c6536069419ce043e4405538c6cae"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-03T16:03:22.339841"
    }
}